src/mpi/init/init_async.c

```diff
@@ -59,22 +78,170 @@ static void progress_fn(void *data)
     return;
 }
 
+#ifdef DO_ASYNC_THREAD_AFFINITY
+MPL_STATIC_INLINE_PREFIX int MPIDI_parse_progress_thread_affinity(int *thread_affinity,
+                                                                  int async_threads_per_node)
+{
+    int th_idx, read_count = 0, mpi_errno = MPI_SUCCESS;
+    char *affinity_copy = NULL;
+    const char *affinity_to_parse = MPIR_CVAR_CH4_PROGRESS_THREAD_AFFINITY;
+    char *proc_id_str, *tmp;
+    size_t proc_count;
+
+    if (!affinity_to_parse || strlen(affinity_to_parse) == 0) {
+        MPIR_ERR_SET1(mpi_errno, MPI_ERR_OTHER, "**parse_thread_affinity",
+                      "**parse_thread_affinity %s", affinity_to_parse);
+        goto fn_fail;
```

Comment:
You can use `MPIR_ERR_SETANDJUMP1`. Same for the reset.

Reply:
Sure.

Reply:
Done